src/base/base.scss

```
@@ -76,10 +76,43 @@ button {
 }
 
 details {
-  summary { cursor: pointer; }
+  summary {
+    cursor: pointer;
+  }
 
   &:not([open]) {
     // Set details content hidden by default for browsers that don't do this
-    > *:not(summary) { display: none !important; }
+    > *:not(summary) {
+      display: none !important;
+    }
+  }
+}
+
+// global focus styles
+
+// fallback :focus state
+:focus,
+.focus {
+  &:is(a, button, input, textarea) {
+    @include focusOutline;
+  }
+}
+
+// remove fallback :focus if :focus-visible is supported
+:focus:is(a, button, input, textarea):not(:focus-visible) {
+  outline: solid 1px transparent;
```

```suggestion
  box-shadow: none;
```

I think until we switch to `outline` (when there is better support in Safari) we have to remove the `box-shadow` otherwise it shows when clicking with the mouse.

https://user-images.githubusercontent.com/378023/141042758-838da6d2-0fb7-4b39-90bc-9168654a6248.mov

---

Thanks for reviewing Simon!

I addressed your feedback. I chatted with @vdepizzol and we feel like it makes sense to just use `outline` _now_ even though Safari won't have a nice border-radius like other browsers. So that takes care of a lot of the issues you were noticing!

The `space` bar thing.. I'm not sure 🤔 I couldn't reproduce that after I switched to `outline` so maybe its fixed now? And the disappearing inputs 😱 not sure what was going on there! Let me know if you still see it on this next build.

---

> The `space` bar thing..

Yes, this is now fixed! 👍 

> And the disappearing inputs 😱 not sure what was going on there! Let me know if you still see it on this next build.

I can still reproduce it with [this docs version](https://primer-css-dbkqalvz4-primer.vercel.app/css/components/forms). But it could also be a Chrome extension or something specific to my setup. Interestingly it's ok in [storybook](https://primer-css-dbkqalvz4-primer.vercel.app/css/storybook?path=/story/components-forms-input--playground). 🤷

---

> Is it weird that when an input and button is combined, the button has a `2px` gap, but the input not:

Given the general depth understood for input fields (recessed/inset) and buttons (raised/elevated) I think that's an expected behavior. :v:

> I can still reproduce it with [this docs version](https://primer-css-dbkqalvz4-primer.vercel.app/css/components/forms).

Hm I'm seeing this too on Chrome (95.0.4638.54, macOS). It's quite weird, I looked it quickly and couldn't spot a clear culprit. It's happening with `<select>` elements as well. 🤔

---

This should  be fixed now, but definitely give it a test!